BackEnd/Data/MySQLMethods.cs

```
@@ -100,50 +99,152 @@ public void Update(User userParam)
+                Tasks = p.Tasks.Select(s => new TaskModel
+                {
+                    TaskId = s.TaskId,
+                    Description = s.Description,
+                    Status = s.Status,
+                    Importance = s.Importance,
+                    UserId = s.UserId ?? default(int),
+                    ProjectId = s.ProjectId
+                })
+            }).FirstOrDefault();
+            return result;
+        }
+
+        public IEnumerable<ProjectModel> AddProject(Project project)
+        {
+            _context.Projects.Add(project);
+            _context.SaveChanges();
+            var result = _context.Projects.Select(p =>
+            new ProjectModel
+            {
+                Id = p.Id,
+                Title = p.Title,
+                Description = p.Description,
+                Tasks = p.Tasks.Select(s => new TaskModel
+                {
+                    TaskId = s.TaskId,
+                    Description = s.Description,
+                    Status = s.Status,
+                    Importance = s.Importance,
+                    ProjectId = s.ProjectId
+                })
+            });
+            return result;
+        }
+
+        public Project EditProject(int id, Project project)
+        {
+            var singleProject = _context.Projects.Find(id);
+            if (singleProject == null)
+                throw new AppException("Project not found");
+
+            if (!string.IsNullOrWhiteSpace(project.Title))
+                singleProject.Title = project.Title;
+
+            if (!string.IsNullOrWhiteSpace(project.Description))
+                singleProject.Description = project.Description;
+
+            _context.Projects.Update(singleProject);
+            _context.SaveChanges();
+
+            return singleProject;
+        }
+
+        public void DeleteProject(int id)
+        {
+            var singleProject = _context.Projects.Find(id);
+            if (singleProject != null)
```

You should return an error if the project does not exists